presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -310,6 +314,22 @@ private boolean isCorrelated(Expression expression)
         {
             return correlation.stream().anyMatch(VariablesExtractor.extractUnique(expression, variableAllocator.getTypes())::contains);
         }
+
+        private class ExpressionEvaluationVisitor
```

Comment:
I would extend DefaultTraversalVisitor instead and only override visitSymbolReference.  that way you'll automatically have support for nested casts and other kinds of expressions as well.

Reply:
sorry that won't quite work

Reply:
maybe do VariablesExtractor.extractAll() or extractUnique() for the right and left sides and then check that each side has exactly one variableReferenceExpression.

Reply:
I replace the current logic of looping through each correlatedConjuncts for correlated symbol mapping by using `VariablesExtractor.extractUnique` to determine any exists in the correlation map. This is a better approach, thanks much for the suggestion!